src/kubernetes_multus.py

```diff
@@ -0,0 +1,276 @@
+            if e.response.status_code == 404:
+                raise KubernetesMultusError(
+                    "NetworkAttachmentDefinitionResource resource not found. "
+                    "You may need to install Multus CNI."
+                )
+            else:
+                raise KubernetesMultusError(
+                    f"Unexpected outcome when retrieving network attachment definition {name}"
+                )
+
+    def create_network_attachment_definition(
+        self, network_attachment_definition: NetworkAttachmentDefinition
+    ) -> None:
+        """Creates a NetworkAttachmentDefinitionResource.
+
+        Args:
+            network_attachment_definition: NetworkAttachmentDefinition object
+        """
+        network_attachment_definition_obj = NetworkAttachmentDefinitionResource(
+            metadata=ObjectMeta(name=network_attachment_definition.name),
+            spec=network_attachment_definition.spec,
+        )
+        self.client.create(obj=network_attachment_definition_obj, namespace=self.namespace)  # type: ignore[call-overload]  # noqa: E501
+        logger.info(
+            f"NetworkAttachmentDefinitionResource {network_attachment_definition.name} created"
+        )
+
+    def delete_network_attachment_definition(self, name: str) -> None:
+        """Deletes network attachment definition based on name.
+
+        Args:
+            name: NetworkAttachmentDefinition name
+        """
+        self.client.delete(
+            res=NetworkAttachmentDefinitionResource, name=name, namespace=self.namespace
+        )
+        logger.info(f"NetworkAttachmentDefinition {name} deleted")
+
+    def patch_statefulset(self, name: str, network_annotations: List[NetworkAnnotation]) -> None:
+        """Patches a statefulset with multus annotation.
+
+        Args:
+            name: Statefulset name
+            network_annotations: List of network annotations
+        """
+        if not network_annotations:
+            logger.info("No network annotations were provided")
+            return
+        statefulset = self.client.get(res=StatefulSet, name=name, namespace=self.namespace)
+        statefulset.spec.template.metadata.annotations["k8s.v1.cni.cncf.io/networks"] = json.dumps(  # type: ignore[attr-defined]  # noqa: E501
+            [network_annotation.dict() for network_annotation in network_annotations]
+        )
+
+        # statefulset.spec.template.spec.containers[2].securityContext.privileged = True
+        # statefulset.spec.template.spec.containers[2].securityContext.capabilities = Capabilities(
+        #     add=[
+        #         "NET_ADMIN",
+        #     ]
+        # )
+
+        self.client.patch(
+            res=StatefulSet,
+            name=name,
+            obj=statefulset,
+            patch_type=PatchType.MERGE,
+            namespace=self.namespace,
+        )
+        logger.info(f"Multus annotation added to {name} Statefulset")
+
+    def statefulset_is_patched(
```

Comment:
This would be clearer with the parameters.
```suggestion
    def statefulset_has_annotations(
```

Reply:
It's more than annotation, it's also that the container security context is modified